addons/web/static/tests/views/list_tests.js

```diff
@@ -9592,6 +9592,40 @@ QUnit.module('Views', {
 
         list.destroy();
     });
+
+    QUnit.test('selecting a row after another one containing a table within an html field should be the correct one', async function (assert) {
+        assert.expect(1);
+
+        this.data.foo.fields.html = {string: "HTML field", type: "html"}
+        this.data.foo.records[0].html = `
+            <table class="table table-bordered">
+                <tbody>
+                    <tr>
+                        <td><br></td>
+                        <td><br></td>
+                    </tr>
+                     <tr>
+                        <td><br></td>
+                        <td><br></td>
+                    </tr>
+                </tbody>
+            </table>`;
+
+        var list = await createView({
+            View: ListView,
+            model: 'foo',
+            data: this.data,
+            arch: '<tree editable="top" multi_edit="1">' +
+                '<field name="html"/>' +
+                '</tree>',
+        });
+
+        await testUtils.dom.click(list.$('table.o_list_table > tbody > tr:eq(1) td.o_field_cell'))
```

Comment:
This click seems a bit too specific, I think clicking of the nth o_data_cell should suffice and reflect a real use case.

also, make sure your test fails when the fix is absent

Reply:
For me, the test fails without the fix